mijia/src/bluetooth.rs

```diff
@@ -64,6 +81,53 @@ impl DeviceId {
     }
 }
 
+/// Opaque identifier for a GATT service on a Bluetooth device.
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct ServiceId {
+    pub(crate) object_path: String,
+}
+
+impl ServiceId {
+    pub(crate) fn new(object_path: &str) -> Self {
+        Self {
+            object_path: object_path.to_owned(),
+        }
+    }
+
+    /// Get the ID of the device on which this service was advertised.
+    pub fn device(&self) -> DeviceId {
+        let index = self
+            .object_path
+            .rfind('/')
+            .expect("ServiceId object_path doesn't contain a slash.");
```

Comment:
```suggestion
            .expect("ServiceId object_path must contain a slash.");
```
and also below.

I find this type of `expect()` message really hard to read, because my brain always reads it as "expect that ServiceId object_path doesn't contain a slash." which is the opposite of what we are actually expecting. 

The code is much easier to read if you state your expectation/invariant in the message as a MUST statement, and the error message is still easy to read if the thing explodes.

("expect" is a very unfortunate name for this function, but this is the world that we live in)

Reply:
Done.